### ISSUE TITLE
Not compatible with pre-EIP155 transactions

### ISSUE THREAD

**Issue by CedarMist** [oasisprotocol/oasis-sdk#1439]
I'm happy if this is considered 'wontfix', but it's worth logging a ticket anyway.

ChainID verification disallows 'any chain' replayable transactions (pre-EIP155).

This differs from the behavior of go-ethereum, and is used by projects like Multicall to allow anybody to submit pre-signed deployment transactions to deploy contracts on any chain at pre-defined addresses without the original author interacting. 

I'm not sure how frequently this is used.

One example: https://github.com/mds1/multicall#new-deployments - they provide pre-signed transaction to submit to your chain.

Note on pre-[EIP155](https://eips.ethereum.org/EIPS/eip-155) transactions:

> If block.number >= FORK_BLKNUM and v = CHAIN_ID * 2 + 35 or v = CHAIN_ID * 2 + 36, then when computing the hash of a transaction for purposes of recovering, instead of hashing six rlp encoded elements (nonce, gasprice, startgas, to, value, data), hash nine rlp encoded elements (nonce, gasprice, startgas, to, value, data, chainid, 0, 0). The currently existing signature scheme using v = 27 and v = 28 remains valid and continues to operate under the same rules as it did previously.

In go-ethereum it checks whether the transaction is replay protected: https://github.com/ethereum/go-ethereum/blob/e13fa32cea3497aa311226c62daf21c5847c8e4f/core/types/transaction.go#L242C6-L259
```go
func isProtectedV(V *big.Int) bool {
	if V.BitLen() <= 8 {
		v := V.Uint64()
		return v != 27 && v != 28 && v != 1 && v != 0
	}
	// anything not 27 or 28 is considered protected
	return true
}

// Protected says whether the transaction is replay-protected.
func (tx *Transaction) Protected() bool {
	switch tx := tx.inner.(type) {
	case *LegacyTx:
		return tx.V != nil && isProtectedV(tx.V)
	default:
		return true
	}
}
```

Then when sanity checking the signature it sets the `v` (recovery id) parameter accordingly, which will invalidate the signature if it's 'protected'.

Whereas in oasis-sdk it always validates the chain ID:

https://github.com/oasisprotocol/oasis-sdk/blob/89d76d7be2c0e15a5d2073984c8fc32d903172f4/runtime-sdk/modules/evm/src/raw_tx.rs#L119-L125

Resulting in an error when the multicall transaction is submitted:

    ProviderError: client: transaction check failed: runtime error: module: core code: 1
    message: malformed transaction: chain ID None, expected Some(23294)

**Comment by CedarMist** [comment]
Have opened a branch for this:

https://github.com/oasisprotocol/oasis-sdk/tree/CedarMist/eip1559-multicall

**Comment by kostko** [comment]
Fixed in #1792.
